packages/mattermost-redux/src/utils/post_list.ts

```diff
@@ -323,142 +322,100 @@ export function makeGenerateCombinedPost(): (state: GlobalState, combinedId: str
     );
 }
 
-export const postTypePriority = {
-    [Posts.POST_TYPES.JOIN_TEAM]: 0,
-    [Posts.POST_TYPES.ADD_TO_TEAM]: 1,
-    [Posts.POST_TYPES.LEAVE_TEAM]: 2,
-    [Posts.POST_TYPES.REMOVE_FROM_TEAM]: 3,
-    [Posts.POST_TYPES.JOIN_CHANNEL]: 4,
-    [Posts.POST_TYPES.ADD_TO_CHANNEL]: 5,
-    [Posts.POST_TYPES.LEAVE_CHANNEL]: 6,
-    [Posts.POST_TYPES.REMOVE_FROM_CHANNEL]: 7,
-    [Posts.POST_TYPES.PURPOSE_CHANGE]: 8,
-    [Posts.POST_TYPES.HEADER_CHANGE]: 9,
-    [Posts.POST_TYPES.JOIN_LEAVE]: 10,
-    [Posts.POST_TYPES.DISPLAYNAME_CHANGE]: 11,
-    [Posts.POST_TYPES.CONVERT_CHANNEL]: 12,
-    [Posts.POST_TYPES.CHANNEL_DELETED]: 13,
-    [Posts.POST_TYPES.CHANNEL_UNARCHIVED]: 14,
-    [Posts.POST_TYPES.ADD_REMOVE]: 15,
-    [Posts.POST_TYPES.EPHEMERAL]: 16,
-};
-
-export function comparePostTypes(a: typeof postTypePriority, b: typeof postTypePriority) {
-    return postTypePriority[a.postType] - postTypePriority[b.postType];
-}
-
-function extractUserActivityData(userActivities: any) {
+function extractUserActivityData(userActivities: ActivityEntry[]) {
     const messageData: any[] = [];
     const allUserIds: string[] = [];
     const allUsernames: string[] = [];
-    Object.entries(userActivities).forEach(([postType, values]: [string, any]) => {
-        if (
-            postType === Posts.POST_TYPES.ADD_TO_TEAM ||
-            postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
-            postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
-        ) {
-            Object.keys(values).map((key) => [key, values[key]]).forEach(([actorId, users]) => {
-                if (Array.isArray(users)) {
-                    throw new Error('Invalid Post activity data');
-                }
-                const {ids, usernames} = users;
-                messageData.push({postType, userIds: [...usernames, ...ids], actorId});
-                if (ids.length > 0) {
-                    allUserIds.push(...ids);
+    userActivities.forEach((activity) => {
+        if (isUsersRelatedPost(activity.postType)) {
```

Comment:
To avoid needing an if-else here, could you make it so that `combineUserActivitySystemPost` always sets `usernames` and `userIds` to something, even if that's an empty array? That way, we should be able to just loop over `userActivities` and and just add all the user IDs to `allUserIds` and all the usernames to `allUsernames` without any branching in the code. It'd also mean that we wouldn't have to throw any errors which would probably crash the whole app based on where this is called from